src/linear_solver_routines/arnoldi_solver.jl

```diff
@@ -0,0 +1,71 @@
+# Date 09/10/2024
+# Author: Christian Varner
+# Purpose: Implement deterministic solver.
+
+"""
+    arnoldi!(x, A, b, k)
```

Comment:
need to specify types here

Reply:
Unrelated to this comment but about tabbing in docstrings for documentation. I looked at [examples](https://documenter.juliadocs.org/stable/man/examples/) that `Documenter.jl` provides and it doesn't seem like you need to indent. I looked at the [first package](https://github.com/Evizero/Augmentor.jl/blob/master/src/augment.jl) they provided, and the source code.